src/App.js

```diff
@@ -1,7 +1,72 @@
-import React from "react";
+import React, { useEffect, useState } from 'react';
+import jwt_decode from 'jwt-decode';
+import Parse from 'parse';
 
-const App = () => {
-  return <h1>Hello React!</h1>;
-};
+function App() {
+  const [ user, setUser ] = useState({});
 
-export default App;
+  function handleCallbackResponse(response) {
+    console.log("Encoded User Token: " + response.credential);
+    var userObject = jwt_decode(response.credential);
+    console.log(userObject);
+    setUser(userObject);
+    document.getElementById("signInDiv").hidden = true;
+
+    // Log in to Parse Server using Google token
+    Parse.User.logInWith('google', { authData: { id: userObject.sub, access_token: response.credential } })
+      .then(user => {
+        console.log('User logged in through Google:', user);
+      })
+      .catch(error => {
+        console.error('Error logging in through Google:', error);
+      });
+  }
+
+  function handleSignOut(event) {
+    // Log out from Parse Server
+    Parse.User.logOut()
+      .then(() => {
+        console.log('User logged out from Parse');
+        setUser({});
+        document.getElementById("signInDiv").hidden = false;
+      })
+      .catch(error => {
+        console.error('Error logging out:', error);
+      });
+  }
+
+  useEffect(() => {
```

Comment:
useEffects should always be at the top of react component, right after the states